### PR TITLE
Update Build Applications Page

### PR DESCRIPTION
**Pull request by emmwalsh**
# Description

There are a number of tiny mistakes which I have fixed ahead of the 2024.2 release as requested in [DOC-12325]

I removed all duplicated product mentions and changed 2 >> Set environment variables by calling <install dir>/setvars.sh.

  to   <<  Set environment variables by calling <install dir>/vars.sh.

### CODE REVIEW

**Comment by emmwalsh** [comment]
FYI @bdmoore1

**Comment by emmwalsh** [comment]
@aepanchi can you check this PR and approve it please? Thanks!

**Review by bdmoore1 — APPROVED** [review]

**Comment by emmwalsh** [comment]
@maria-Petrova can you approve and merge this PR please?

**Review by maria-Petrova — APPROVED** [review]
Need to double-check the list of the Windows libs.

File: docs/source/onedal/build_app/build-application.rst [hunk 1]
```diff
@@ -117,13 +117,11 @@ Applications on Windows* OS
 
       * - Static linking
         - 
-          | onedal_core.lib,
           | onedal_core.lib,
           | onedal_thread.lib
       * - Dynamic linking
         - 
           | onedal_core_dll.lib
-          | onedal_core_dll.lib
```
Comment (maria-Petrova) [inline docs/source/onedal/build_app/build-application.rst:126]: @napetrov, should it be replaced with **onedal_dpc_dll.lib**?

**Review by maria-Petrova — CHANGES_REQUESTED** [review]

File: docs/source/onedal/build_app/build-application.rst (hunk 1 above)
Comment (napetrov) [reply docs/source/onedal/build_app/build-application.rst:126]: It's just an example here.  ether can work

**Review by maria-Petrova — APPROVED** [review]

**Pull request merged**